tools/osbuild-mpp

```diff
@@ -1406,28 +1409,40 @@ class ManifestFileV2(ManifestFile):
         def embed_data(ip, mpp):
             uid = mpp["id"]
             path = mpp.get("path")
+            url = mpp.get("url")
             text = mpp.get("text")
 
-            if path and text:
-                raise ValueError(f"Cannot specify both 'path' and 'text' for '{uid}'")
+            input_count = (1 if text else 0) + (1 if path else 0) + (1 if url else 0)
+            if input_count == 0:
+                raise ValueError(f"At least one of 'path', 'url' or 'text' must be specified for '{uid}'")
+            if input_count > 1:
+                raise ValueError(f"Only one of 'path', 'url' or 'text' may be specified for '{uid}'")
 
             if path:
                 f, _ = self.find_and_open_file(path, [], mode="rb")
                 with f:
                     data = f.read()
+            elif url:
+                response = urllib.request.urlopen(url)
+                data = response.fp.read()
```

Comment:
You fetch the data but you never use it, right? Is this to verify the request is valid? Or am I reading this wrong?

Reply:
The data is used to calculate the checksum (See line 1431: `checksum = hashlib.sha256(data).hexdigest()`)